src/lib.rs

```diff
@@ -543,6 +558,95 @@ fn read_into_buffer(fd: RawFd, buffer: &mut [u8]) -> isize {
     }
 }
 
+// Use this when 1.24 is available for use. We can use the hard-coded 16 due to Linux's ABI
+// guarantees.
+// const EVENT_MAX_SIZE: usize = mem::size_of::<ffi::inotify_event>() + (FILENAME_MAX as usize) + 1;
+const EVENT_MAX_SIZE: usize = 16 + (FILENAME_MAX as usize) + 1;
```

Comment:
Hmm, 1.24 is out now, but maybe it's better to leave this as is for the time being. Maybe some people will be stuck with older Rust versions on some Linux distributions. Any opinions? I don't feel strongly either way.

In one of your comments, you mentioned the possibility of making the value of this constant configurable via a Cargo feature, but I think this is fine as it is. It's a sane default, and if people have a problem with it they can file an issue. Better that, than front-loading the complexity for a solution that maybe nobody cares about.

Reply:
I'd like to leave the minimum version as low as possible to start.